DDF 1172 Unit tests for admin-core-appservice pkg

NGoss opened the pull request:
@brendan-hofmann @mverret

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf-platform/88)
<!-- Reviewable:end -->

brendan-hofmann commented:
Don't forget to bump up the coverage ratios in the pom.xml to reflect your improvements.

---

Reviewed 19 of 20 files at r1, 1 of 1 files at r2, 1 of 21 files at r3.
Review status: 2 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java, line 43 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/88#-Jv0A0IYZCmioRuGdRHW)** ([raw file](https://github.com/codice/ddf-platform/blob/1417c6c56c3c40d3f6763342b383f3d18fcfdc51/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java#L43)):</sup>
This will need to be fixed for both tests.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)
<!-- Sent from Reviewable.io -->

NGoss commented:
Updated the pom to reflect coverage.

---

Review status: 2 of 24 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java, line 45 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/88#-Jv0A0IYZCmioRuGdRHW)** ([raw file](https://github.com/codice/ddf-platform/blob/1417c6c56c3c40d3f6763342b383f3d18fcfdc51/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java#L45)):</sup>
Fixed with the getter/setter solution.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)
<!-- Sent from Reviewable.io -->

brendan-hofmann commented:
+1

---

Reviewed 19 of 21 files at r3, 2 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)
<!-- Sent from Reviewable.io -->

mverret commented:
Reviewed 1 of 1 files at r2, 20 of 21 files at r3, 2 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)
<!-- Sent from Reviewable.io -->

mverret commented:
I was unable to do a build.  The code coverage in both Java 7 and Java 8 were .01 off.  If you lower the expected minimums by .01 then it should be able to build.

mverret commented:
Changed the pom file in my local checkout and verified that it does in fact build and all tests run successfully.  Doing a full build of all of DDF now that will hopefully complete over the weekend.

mverret commented:
I actually stayed.  Full build and tests were successful with the lowered coverage numbers.  Please make sure to update them.

NGoss commented:
Lowered the coverage numbers by .01 each. Builds fine on my local machine now.

brendan-hofmann commented:
Verified successful build with adjusted coverage. :+1: 

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)
<!-- Sent from Reviewable.io -->

brendan-hofmann commented on admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java:
```diff
@@ -0,0 +1,114 @@
+
+            applicationUploadEndpoint
+                    .setDefaultFileLocation("target/ApplicationUploadEndpointTest");
+            assertNotNull(applicationUploadEndpoint.update(testMultipartBody, testUriInfo));
```
What type of response are you expecting here? Could assert that it's a success/fail/whatever instead of just not null.

For future reference we tend to like assertThat() better as well, since it produces much more readable test output.

I'm not going to comment on every instance of assertNotNull(), but the same general ideas apply to most of them. ;)

brendan-hofmann commented on admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/impl/AddApplicationCommandTest.java:
```diff
@@ -0,0 +1,61 @@
+            verify(testAppService).addApplication(any(URI.class));
+        } catch (Exception e) {
+            logger.info("Exception: ", e);
+            fail();
```
Instead of failing in a try/catch, you could just add the exception to the method signature and let junit fail the test if the exception is thrown. It's less verbose and a little cleaner.

Again, not going to comment on every instance, but this principle is generally applicable.

brendan-hofmann commented on admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/impl/ZipFileApplicationDetailsTest.java:
```diff
@@ -0,0 +1,38 @@
+     */
+    @Test
+    public void testConstructor() {
+        ZipFileApplicationDetails testZipFile = new ZipFileApplicationDetails("TestName", "0.0.0");
```
Another general comment; test strings, especially test strings being asserted against, could be defined as constants instead. This makes the test less fragile to future changes and reduces duplication.

brendan-hofmann commented:
Currently, most of the uncovered lines look to be inside fail state branches. This isn't unique to this package, but we could use more error handling test coverage in general, especially in cases where there's an expectation of graceful recovery or continuing to function. This is also a good opportunity to think critically about our edge cases.

NGoss commented:
Fixed the issues with the test strings and assertNotNull calls, and did what I could to improve fail state coverage. Let me know of any further issues (if any).

---

Review status: 2 of 30 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)
<!-- Sent from Reviewable.io -->

brendan-hofmann commented:
Since there are a lot of commits in this PR, you might want to consider squashing them in some logical fashion.

---

Reviewed 28 of 28 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

<sup>**[admin/core/admin-core-appservice/pom.xml, line 127 [r7]](https://reviewable.io:443/reviews/codice/ddf-platform/88#-Jvpgkqt9_3OVTXp0DK0)** ([raw file](https://github.com/codice/ddf-platform/blob/9a2d5242f19266a39758dbcfde6a6191452532a5/admin/core/admin-core-appservice/pom.xml#L127)):</sup>
Way to meet or exceed our minimum coverage goals. :+1: 

---

<sup>**[admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java, line 128 [r7]](https://reviewable.io:443/reviews/codice/ddf-platform/88#-JvphFLyR35NdvPhufHO)** ([raw file](https://github.com/codice/ddf-platform/blob/9a2d5242f19266a39758dbcfde6a6191452532a5/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java#L128)):</sup>
Checking response codes instead of message content, good! :+1: 

---

<sup>**[admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/impl/AddApplicationCommandTest.java, line 52 [r7]](https://reviewable.io:443/reviews/codice/ddf-platform/88#-JvphlD7io4-IJRhCzr1)** ([raw file](https://github.com/codice/ddf-platform/blob/9a2d5242f19266a39758dbcfde6a6191452532a5/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/impl/AddApplicationCommandTest.java#L52)):</sup>
It doesn't really matter as much for unit tests, but in general, catching or throwing "Exception" should be avoided, and typed exceptions should be preferred. 

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)

<!-- Sent from Reviewable.io -->

NGoss commented:
Squashed the commits down to three.

---

Review status: 30 of 31 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/88)

<!-- Sent from Reviewable.io -->

NGoss commented:
DDF-platform merged into DDF.
New pull request: https://github.com/codice/ddf/pull/90
